tests/group_functions/group_scan.h

```diff
@@ -416,11 +416,9 @@ void check_scan_over_group(sycl::queue& queue, sycl::range<D> range, OpT op,
                 sycl::group<D> group = item.get_group();
                 sycl::sub_group sub_group = item.get_sub_group();
 
-                // Use the local id of the item in the group to place results of
-                // the scan operation in the order of the items.
-                auto g_index = group.get_group_linear_id() *
-                                   group.get_local_linear_range() +
-                               group.get_local_linear_id();
+                // Use the global linear id of the item in the group to place
+                // results of the scan operation in the order of the items.
+                auto g_index = item.get_global_linear_id();
```

Comment:
I don't think this is an option. I believe this is exactly what we moved away from a couple of patches ago. The problem is that partial results of scan appear in the order of the items in the sub-group, which is not required to follow the order of the global ID.

We can use it, but then we will also have to store the order of the sub-group items and use that when verifying.

Reply:
Thanks. I've read your previous PR #804. To make sure we are on the same page:
1. [SYCL spec - group class](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#group-class) requires both the **group ordering** and the **items ordering within each group**, to be consistent with the [linearized dimension](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:multi-dim-linearization), so using `item.get_global_linear_id()` should be equivalent to `group.get_group_linear_id() * group.get_local_linear_range() + group.get_local_linear_id()` for work-groups.
2. However, [SYCL spec - Work-group data parallel kernels](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_work_group_data_parallel_kernels) gives us no guarantees on sub-group ordering: 
> Portable device code must not assume that work-items within a sub-group execute in any particular order, that work-groups are subdivided into sub-groups in a specific way, nor that the work-items within a sub-group provide specific forward progress guarantees.

For the sub-group part, I think we need to store both the `sub_group.get_group_linear_id()` (as an abstract identifier of the sub-group) and the `sub_group.get_local_linear_id()` (as an identifier for the item within the sub-group) and then collect and reorder the reference input data for each sub-group when verifying. So that we permit the implementation the maximal freedom of sub-group partitioning and ordering.